xclim/subset.py

```diff
@@ -781,16 +796,57 @@ def in_bounds(bounds: Tuple[float, float], coord: xarray.Coordinate) -> bool:
 
 
 def _check_desc_coords(coord, bounds, dim):
-    """If dataset coordinates are descending reverse bounds"""
+    """If Dataset coordinates are descending reverse bounds"""
     if np.all(coord.diff(dim=dim) < 0):
         bounds = np.flip(bounds)
     return bounds
 
 
+def _check_has_overlaps(polygons: gpd.GeoDataFrame):
+    non_overlapping = []
+    for n, p in enumerate(polygons["geometry"][:-1], 1):
+        if not any(p.overlaps(g) for g in polygons["geometry"][n:]):
+            non_overlapping.append(p)
+    if len(polygons) == len(non_overlapping):
+        return
+    return False
```

Comment:
Shouldn't we issue a warning here? The output of the this is not checked above, so I'm not sure what it really does?

Reply:
On it right now.